akka-remote/src/main/scala/akka/remote/serialization/MiscMessageSerializer.scala

```diff
@@ -106,6 +120,14 @@ class MiscMessageSerializer(val system: ExtendedActorSystem) extends SerializerW
   private def deserializeActorRef(actorRef: ContainerFormats.ActorRef): ActorRef =
     serialization.system.provider.resolveActorRef(actorRef.getPath)
 
+  private def deserializeOption(bytes: Array[Byte]): Option[Any] = {
```

Comment:
one small optimization would be to check `bytes.length == 0` as the very first thing for `None`, it saves one allocation of `ContainerFormats.Option`